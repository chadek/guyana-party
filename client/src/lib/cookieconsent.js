import 'cookieconsent/build/cookieconsent.min.css'

if (typeof window !== 'undefined') {
  // Import non server side rendering cookieconsent
  import('cookieconsent').then(() => {
    window.cookieconsent.initialise({
      palette: {
        popup: { background: '#fff' },
        button: { background: '#3f51b5' }
      },
      theme: 'edgeless',
      position: 'bottom-left',
      content: {
        message:
          "Nous utilisons des cookies afin d'améliorer votre expérience sur le site et ainsi vous proposer un contenu adapté.",
        link: 'En savoir plus',
        href: '/mentions-legales/#cookies',
        dismiss: 'OK'
      }
    })
  })
}
