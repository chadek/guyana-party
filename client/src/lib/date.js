import { zonedTimeToUtc, utcToZonedTime, format } from 'date-fns-tz'
import fr from 'date-fns/locale/fr'
import tzList from './tzList'

export const userTZ = () => {
  try {
    let offset = new Date().getTimezoneOffset()
    const o = Math.abs(offset)
    offset = `${(offset < 0 ? '+' : '-') + `00${Math.floor(o / 60)}`.slice(-2)}:${`00${o % 60}`.slice(-2)}`
    return tzList.filter(t => t.offset === offset)[0].tzCode
  } catch (error) {
    console.log(error)
    return 'Europe/London'
  }
}

export const toUTCIsoDate = (date, timeZone) => zonedTimeToUtc(date, timeZone).toISOString()

export const toZonedTime = (date, timeZone) => utcToZonedTime(date, timeZone)

export const dateToStr = (date, timeZone, formatStr = 'd MMM yyyy à HH:mm (zzz)') => {
  const locale = fr
  if (timeZone) return format(utcToZonedTime(date, timeZone), formatStr, { locale, timeZone })
  return format(new Date(date), formatStr, { locale })
}

export const formatStart = startDate => {
  const tz = userTZ()
  const d = dateToStr(startDate, tz, 'EEEE d MMMM yyyy').split(' ')
  let day = d[0]
  day = day.charAt(0).toUpperCase() + day.slice(1)
  const dayNb = d[1]
  let month = d[2]
  month = month.charAt(0).toUpperCase() + month.slice(1)
  const year = d[3]
  return `${day} ${dayNb} ${month} ${year}`
}

export const formatPlage = ({ startDate, endDate }, showUTC = false) => {
  const tz = userTZ()
  const start = dateToStr(startDate, tz, 'dd MM HH:mm (zzz)').split(' ')
  const end = dateToStr(endDate, tz, 'dd MM HH:mm').split(' ')
  const utc = showUTC ? ` ${start[3]}` : ''
  if (start[0] + start[1] === end[0] + end[1]) {
    return `Le ${start[0]}/${start[1]} de ${start[2]} à ${end[2]}${utc}`
  }
  return `Du ${start[0]}/${start[1]} à ${start[2]} au ${end[0]}/${end[1]} à ${end[2]}${utc}`
}

export const days = [
  { label: 'Lundi', value: 'mon' },
  { label: 'Mardi', value: 'tue' },
  { label: 'Mercredi', value: 'wed' },
  { label: 'Jeudi', value: 'thu' },
  { label: 'Vendredi', value: 'fri' },
  { label: 'Samedi', value: 'sat' },
  { label: 'Dimanche', value: 'sun' }
]
