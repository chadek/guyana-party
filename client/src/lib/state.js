import { State, subscribe } from 'jstates-react'

export const snackState = new State('snackState', { msg: '', variant: '' })

export { subscribe }
