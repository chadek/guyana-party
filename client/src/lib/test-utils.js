import React from 'react'
import PropTypes from 'prop-types'
import { render } from 'enzyme'

import { ThemeProvider } from 'styled-components'
import { theme, Style } from '../components/styles/Style'

const AllTheProviders = ({ children }) => (
  <ThemeProvider theme={theme}>
    <>
      <Style />
      {children}
    </>
  </ThemeProvider>
)

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired
}

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
// export * from 'enzyme'
export { shallow } from 'enzyme'
export { expect } from 'chai'

// override render method
export { customRender as render }
