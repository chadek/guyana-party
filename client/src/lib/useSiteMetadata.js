import { useStaticQuery, graphql } from 'gatsby'

export default () => {
  const { site } = useStaticQuery(
    graphql`
      query SiteMetaData {
        site {
          siteMetadata {
            title
            titleTemplate
            description
            author
            keywords
          }
        }
      }
    `
  )
  return site.siteMetadata
}
