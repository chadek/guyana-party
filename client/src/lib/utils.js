import axios from 'axios'
import Cookies from 'js-cookie'
import dompurify from 'dompurify'
import Showdown from 'showdown'
import md5 from 'md5'
import cogoToast from 'cogo-toast'

export const MISSING_TOKEN_ERR = 'Token de connexion requis'

export const reload = () => {
  if (typeof window !== 'undefined') window.location.reload()
}

export const gravatar = email => `https://www.gravatar.com/avatar/${md5(email)}?d=retro`

export const purify = async dirty => dompurify.sanitize(dirty)

export const markToSafeHTML = markdown => {
  const converter = new Showdown.Converter({
    tables: true,
    simplifiedAutoLink: true,
    strikethrough: true,
    tasklists: true
  })
  return purify(converter.makeHtml(markdown))
}

export const getUID = () => Cookies.get('gp_uid')

export const getToken = () => ({ jwt: Cookies.get('gp_jwt'), uid: getUID() })

export const fetcher = url => axios.get(url).then(r => r.data)

export const axiosGet = (url, next, fallback) =>
  axios
    .get(url)
    .then(payload => next(payload))
    .catch(fallback)

export const axiosPost = ({ url, data }, next, fallback) => {
  const jwt = Cookies.get('gp_jwt')
  if (!jwt) return fallback(MISSING_TOKEN_ERR)

  const options = { headers: { authorization: `bearer ${jwt}` } }

  return axios.post(url, data, options).then(next).catch(fallback)
}

export const axiosPut = ({ url, data }, next, fallback) => {
  const jwt = Cookies.get('gp_jwt')
  if (!jwt) return fallback(MISSING_TOKEN_ERR)

  const options = { headers: { authorization: `bearer ${jwt}` } }

  return axios.put(url, data, options).then(next).catch(fallback)
}

export const axiosDelete = (url, next, fallback) => {
  const jwt = Cookies.get('gp_jwt')
  if (!jwt) return fallback(MISSING_TOKEN_ERR)

  const options = { headers: { authorization: `bearer ${jwt}` } }

  return axios.delete(url, options).then(next).catch(fallback)
}

/* Convert the coordinate (lat or lng) to DMS (degrees minutes seconds). */
const toDMS = coord => {
  const absolute = Math.abs(coord)
  const degrees = Math.floor(absolute)
  const minutesNotTruncated = (absolute - degrees) * 60
  const minutes = Math.floor(minutesNotTruncated)
  const seconds = Math.floor((minutesNotTruncated - minutes) * 60)
  return `${degrees}° ${minutes}′ ${seconds}″`
}

export const gpsCoords = (lat, lng) => {
  const northSouth = `${toDMS(lat)} ${Math.sign(lat) >= 0 ? 'N' : 'S'}`
  const eastWest = `${toDMS(lng)} ${Math.sign(lng) >= 0 ? 'E' : 'W'}`
  return `${northSouth}, ${eastWest}`
}

export const compress = (files, next) => {
  if (typeof window === 'undefined') return // window not defined in ssr
  const Compress = require('client-compress') // eslint-disable-line global-require
  const compressor = new Compress({ targetSize: 1.0, quality: 0.75 })
  compressor.compress(files).then(data => next(data))
}

export const scrollTo = selector => {
  const target = document.querySelector(selector)
  if (target) target.scrollIntoView({ behavior: 'smooth', block: 'end' })
}

export const toast = (msg, variant = 'info', heading = '') => {
  const { hide } = cogoToast[variant](msg, {
    hideAfter: 5,
    heading: heading || (variant === 'error' ? 'Erreur !' : ''),
    onClick() {
      hide()
    }
  })
}
