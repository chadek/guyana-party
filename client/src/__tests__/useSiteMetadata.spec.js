import { renderHook } from '@testing-library/react-hooks'
import { expect } from '../lib/test-utils'

import useSiteMetadata from '../lib/useSiteMetadata'

describe('useSiteMetadata', () => {
  it('should match its reference snapshot', () => {
    const { result } = renderHook(() => useSiteMetadata())
    expect(result).to.matchSnapshot()
  })
})
