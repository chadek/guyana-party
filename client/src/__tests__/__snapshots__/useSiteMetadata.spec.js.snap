// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useSiteMetadata should match its reference snapshot 1`] = `
Object {
  "current": Object {
    "author": "Christopher Servius",
    "description": "Une description.",
    "keywords": Array [
      "some",
      "keywords",
    ],
    "title": "Site Title",
  },
  "error": undefined,
}
`;
